CASMNET-1022-12 (#1828)
Cray-HPE/docs-csm: operations/validate_csm_health.md

<a name="booting-csm-barebones-image"></a>

### 4.2 Internal SSH access test execution

The internal SSH access tests may be run on any NCN with the `docs-csm` RPM installed. For details on installing the `docs-csm` RPM,
see [Check for Latest Documentation](../update_product_stream/index.md#documentation).

Execute the tests by running the following command:

```bash
ncn# /usr/share/doc/csm/scripts/operations/pyscripts/start.py test_bican_internal
```

> By default, SSH access will be tested between nodes of type `ncn_master`, `cn`, `uan`, `spine_switch` on all relevant networks. You

Reviewer: ```suggestion
By default, SSH access will be tested between master nodes , compute nodes, UANs, and spine switches. on all relevant networks.
```

> can customize which nodes you want to test from and to, as well as the networks, by following the command line parameters as

Reviewer: ```suggestion
It is possible to customize which nodes and networks will be tested. See the test usage statement for details.
```

> described in the help:

Reviewer: ```suggestion
The script usage statement is displayed by calling the test with the `--help` argument:
```

```bash
ncn# /usr/share/doc/csm/scripts/operations/pyscripts/start.py test_bican_internal --help
```

The test will complete with an overall pass/failure status such as the following:

```text
Overall status: PASSED (Passed: 40, Failed: 0)
```

### 4.3 External SSH access test execution

The external SSH access tests may be run on any system external to the cluster.

1. `python3` must be installed (if it is not already).

1. Obtain the test code.

   There are two options for doing this:

    - Install the `docs-csm` RPM.

      See [Check for Latest Documentation](../update_product_stream/index.md#documentation).

    - Copy over the following folder from a system where the `docs-csm` RPM is installed:

        - `/usr/share/doc/csm/scripts/operations/pyscripts`

> 1. Install the python dependencies

Reviewer: ```suggestion
1. Install the Python dependencies
```

>    Switch to the `pyscripts` working directory and run the following command to install the required python dependencies:

Reviewer: ```suggestion
   Run the following command from the `pyscripts` directory in order to install the required Python dependencies:
```

    ```bash
    external:/usr/share/doc/csm/scripts/operations/pyscripts# pip install .
    ```

> 1. Obtain the admin client secret.

Reviewer: ```suggestion
1. Obtain the `admin` client secret.
```

>    Because we do not have access to `kubectl` outside of the cluster, obtain the admin client secret by running the

Reviewer: ```suggestion
   Because `kubectl` will not work outside of the cluster, obtain the `admin` client secret by running the
```

   following command on an NCN.

    ```bash

>     ncn# kubectl get secrets admin-client-auth -o jsonpath='{.data.client-secret}' | base64 -d

Reviewer: ````suggestion
    ncn# kubectl get secrets admin-client-auth -o jsonpath='{.data.client-secret}' | base64 -d
    ```

    Example output:

    ```text
````

    26947343-d4ab-403b-14e937dbd700
    ```

> 1. In the external system, execute the tests by running the following command:

Reviewer: ```suggestion
1. On the external system, execute the tests.
```

    ```bash
    external:/usr/share/doc/csm/scripts/operations/pyscripts# ./start.py test_bican_external
    ```

> 1. The test will first ask you for the system domain. It will then ask for the admin client secret where you can enter the

Reviewer: ```suggestion
1. When prompted by the test, enter the system domain and the `admin` client secret .
```

>    secret that you obtained from the step above in the NCN. The tests will then run.

Reviewer: ```suggestion
```

>    By default, external SSH access will be tested to nodes of type `ncn_master`, `cn`, `uan`, `spine_switch` on all relevant networks. You

Reviewer: ```suggestion
   By default, SSH access will be tested between master nodes , compute nodes, UANs, and spine switches. on all relevant networks.
```

>    can customize which nodes you want to test, as well as the networks to test on, by following the command line parameters as

Reviewer: ```suggestion
   It is possible to customize which nodes and networks will be tested. See the test usage statement for details.
```

>    described in the help:

Reviewer: ```suggestion
   The script usage statement is displayed by calling the test with the `--help` argument:
```
